### DISCUSSION TITLE
GPURenderPassDescriptor.maxDrawCount seems arcane

### DISCUSSION THREAD

**Discussion by juj** [gpuweb/gpuweb#3376] [Ideas/Feedback]
In the current wording of a recently added field (#3005) https://www.w3.org/TR/webgpu/#dom-gpurenderpassdescriptor-maxdrawcount, it reads

"
The maximum number of draw calls that will be done in the render pass. Used by some implementations to size work injected before the render pass. Keeping the default value is a good default, unless it is known that more draw calls will be done.
"

This leads to developers naturally asking themselves, what is the drawback of increasing this value? Or conversely (and more practically), what is the benefit of reducing this value, if they aren't doing 50 million draw calls per pass?

If I understood https://github.com/gpuweb/gpuweb/issues/2189#issuecomment-1083678053 correctly, an implementation will need to retain a GPU buffer of size `maxDrawCount * 5 * sizeof(uint32_t)` of space for a validation buffer? (that does not quite seem right as it is a huge number of bytes)

So reducing this value they would be able to free up GPU resources for more actual geometry and texture data?

**Comment by Kangz** [comment]
This value is used to be able to size some `dispatchIndirect` or `drawIndirect` calls that will do index-buffer bounds check on some implementations (Safari's). The idea is that the default value is fine to use and not a pessimization (as in smaller values larger than 0 will result in the same command stream being sent to the GPU?). The size of the buffer can be determined afterwards (with argument buffers if I understand correctly) but not the calls themselves.

Basically don't worry about this value and never touch it, things will be ok.

**Reply by juj** [reply]
> Basically don't worry about this value and never touch it, things will be ok.

If this is a value that users should not be touching, can it just be deleted from the spec?

**Reply by Kangz** [reply]
That possibility was raised in discussions in the group but the compromise was to keep the value with that very very large default.
